src/Charcoal/Cache/ServiceProvider/CacheServiceProvider.php

```diff
@@ -267,6 +267,17 @@ public function registerService(Container $container)
 
             return new CachePoolFacade($args);
         };
+
+        /**
+         * Provides a way to add custom processing over the cache key for the cache middleware.
+         *
+         * @return Closure
+         */
+        $container['cache/process-cache-key-callback'] = function () {
+            return function ($key) {
+                return $key;
+            };
+        };
```

Comment:
Rename service parameter to `cache/middleware/process-cache-key-callback`.
Maybe remove this and implement the suggestion in `CacheMiddleware.php` where if the callback is missing, return the key as is.